static/skin/index.js

```diff
@@ -1,31 +1,65 @@
 const root = $(`link[type='root']`).attr('href');
+let viewPortHeight = window.innerHeight;
+let isFetching = false;
+let isEnd = false;
+const basicConfig = {
+    'start': 0,
+    'count': Math.floor(viewPortHeight/100 + 1) * 3
+};
+
+function queryUrlBuilder() {
+    let url = `${root}/catalog/search?`;
+    Object.keys(basicConfig).forEach((key, idx) => {
+        url += `${key}=${basicConfig[key]}${idx !== Object.keys(basicConfig).length - 1 ? '&' : ''}`;
+    });
+    return url;
+}
 
 function getInnerHtml(node, query) {
     return node.querySelector(query).innerHTML;
 }
 
-async function loadAndDisplayBooks() {
-  fetch(`${root}/catalog/search`).then(async (resp) => {
-    const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
-    const books = data.querySelectorAll('entry');
-    let bookHtml = '';
-    books.forEach((book) => {
-        const link = book.querySelector('link').getAttribute('href');
-        const title =  getInnerHtml(book, 'title');
-        const description = getInnerHtml(book, 'summary');
+async function loadAndDisplayBooks(append = false) {
+    isFetching = true;
+    fetch(queryUrlBuilder()).then(async (resp) => {
+        const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
+        const books = data.querySelectorAll('entry');
+        let bookHtml = '';
+        books.forEach((book) => {
+            const link = book.querySelector('link').getAttribute('href');
+            const title =  getInnerHtml(book, 'title');
+            const description = getInnerHtml(book, 'summary');
 
-        bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
-            <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
-            <div class='book__title' title='${title}'>${title}</div>
-            <div class='book__description' title='${description}'>${description}</div>
-            <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
-            </div>
-        </div></a>`;
+            bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
+                <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
+                <div class='book__title' title='${title}'>${title}</div>
+                <div class='book__description' title='${description}'>${description}</div>
+                <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
+                </div>
+            </div></a>`;
+        });
+        document.querySelector('.book__list').innerHTML = (append ? document.querySelector('.book__list').innerHTML : '') + bookHtml;
```

Comment:
Do you really need the `append` flag? Can't you just always append? On load you will append to the initial empty content with the same net result.

Reply:
@veloman-yunkan I added this because earlier I used append or replace in case of filters but this won't matter now as using filter has changed most of the implementation and I will be always appending code.